sdk/java/dagger-java-annotation-processor/src/main/java/io/dagger/annotation/processor/DaggerModuleAnnotationProcessor.java

```diff
@@ -0,0 +1,454 @@
+                        return functionInfo;
+                      })
+                  .toList();
+          annotatedObjects.add(
+              new ObjectInfo(
+                  name,
+                  qName,
+                  parseTypeDescription(typeElement),
+                  functionInfos.toArray(new FunctionInfo[functionInfos.size()])));
+        }
+      }
+    }
+
+    return new ModuleInfo(
+        moduleDescription, annotatedObjects.toArray(new ObjectInfo[annotatedObjects.size()]));
+  }
+
+  static JavaFile generate(ModuleInfo moduleInfo) {
+    try {
+      var rm =
+          MethodSpec.methodBuilder("register")
+              .addModifiers(Modifier.PRIVATE)
+              .returns(ModuleID.class)
+              .addException(ExecutionException.class)
+              .addException(DaggerQueryException.class)
+              .addException(InterruptedException.class)
+              .addCode("$T module = dag.module()", io.dagger.client.Module.class);
+      if (isNotBlank(moduleInfo.description())) {
+        rm.addCode("\n    .withDescription($S)", moduleInfo.description());
+      }
+      for (var objectInfo : moduleInfo.objects()) {
+        rm.addCode("\n    .withObject(")
+            .addCode("\n        dag.typeDef().withObject($S", objectInfo.name());
+        if (isNotBlank(objectInfo.description())) {
+          rm.addCode(
+              ", new $T.WithObjectArguments().withDescription($S)",
+              TypeDef.class,
+              objectInfo.description());
+        }
+        rm.addCode(")"); // end of dag.TypeDef().withObject(
+        for (var fnInfo : objectInfo.functions()) {
+          rm.addCode("\n            .withFunction(")
+              .addCode("\n                dag.function($S,", fnInfo.name())
+              .addCode("\n                    ")
+              .addCode(typeDef(fnInfo.returnType()))
+              .addCode(")");
+          if (isNotBlank(fnInfo.description())) {
+            rm.addCode("\n                    .withDescription($S)", fnInfo.description());
+          }
+          for (var parameterInfo : fnInfo.parameters()) {
+            rm.addCode("\n                    .withArg($S, ", parameterInfo.name())
+                .addCode(typeDef(parameterInfo.type()))
+                .addCode(".withOptional($L)", parameterInfo.optional());
+            boolean hasDescription = isNotBlank(parameterInfo.description());
+            boolean hasDefaultValue =
+                parameterInfo.optional() && isNotBlank(parameterInfo.defaultValue());
+            if (hasDescription || parameterInfo.optional()) {
+              rm.addCode(", new $T.WithArgArguments()", io.dagger.client.Function.class);
+              if (hasDescription) {
+                rm.addCode(".withDescription($S)", parameterInfo.description());
+              }
+              if (hasDefaultValue) {
+                rm.addCode(
+                    ".withDefaultValue($T.from($S))", JSON.class, parameterInfo.defaultValue());
```

Comment:
when the parameter type is String maybe we could handle a special case in order to accept non-escaped default values in Optional annotation.

having to write `@Optional(defaultValue="\"some string\"")` rather `@Optional(defaultValue="some string")` is cumbersome

WDYT?

Reply:
Yeah, I don't know how much we should escape from the json string.
It's sad we can't have an `Object` for the value.
But yes, maybe we can check if it's a string and not starting by `"` or `'` and in that case add them to the string automatically.

Reply:
I've done something on that, that should cover the main cases with strings.